### PR TITLE
feat(blockDevices/tags): allows specify Tags for all the block devices

### PR DESCRIPTION
**Pull request by jorgebee65**
This feature allows specify Tags for all the block devices using **blockDevicesTags** on the Pipeline JSON format.

```
      "clusters": [
        {
          "account": "aws-psdev",
          "application": "awsk8",
          "blockDevices": [
            {
              "deleteOnTermination": true,
              "deviceName": "/dev/sda1",
              "size": 8,
              "volumeType": "gp2"
            }
          ],
          "blockDeviceTags": {
            "Creator": "Armory"
          }
        ...
        }
```
Image 2021-02-11 at 1 47 01 PM

### CODE REVIEW

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/services/LaunchTemplateService.java [hunk 1]
```diff
@@ -297,6 +301,16 @@ private RequestLaunchTemplateData buildLaunchTemplateData(
                   .collect(Collectors.toList()));
     }
 
+    if (!asgConfig.getBlockDevicesTags().isEmpty()) {
```
Comment (ajordens) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/services/LaunchTemplateService.java:304]: Might be preferable if this was an implementation of the `AmazonResourceTagger` interface. 

I recently added that to also support ebs volume tagging. The interface may need to change to support passing in the `AsgConfig`.

Comment (jorgebee65) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/services/LaunchTemplateService.java]: I created a `CustomAmazonResourceTagger` class and use it in `AsgConfiguration` (AsgConfig)

**Comment by ajordens** [comment]
Sorry for the delay here ... I pulled this PR locally and tried to make a few changes but for one reason or another it didn't work to push the changes back here.

I think this (https://github.com/spinnaker/clouddriver/pull/5289/files) would be a little cleaner, it extends the existing block device tagger but I think does so in a reasonable way.

**Comment by jorgebee65** [comment]
@ajordens I made the suggested changes.

**Review by ajordens — APPROVED** [review]

**Comment by kinnonyee-armory** [comment]
The following change does not work for 1.26.x.  AWS advises using Launch Templates to define volumes instead. 
https://spinnaker.io/docs/setup/other_config/server-group-launch-settings/aws-ec2/launch-templates/

**Pull request merged**